app/src/main/java/org/mozilla/focus/navigation/ScreenNavigator.java

```diff
@@ -207,6 +209,11 @@ public boolean canGoBack() {
         return result;
     }
 
+    public LiveData<String> getNavigationState() {
+        return Transformations.map(transactionHelper.getTopFragmentState(),
+                fragmentTag -> fragmentTag.isEmpty() ? BROWSER_FRAGMENT_TAG : fragmentTag);
```

Comment:
Currently any operation that adds an entry to back stack MUST do it via ScreenNavigator, otherwise, an exception will always be thrown when the back stack is changed. Empty entry tag is also forbidden. The only case we get empty string here is when there's no record in the back stack. Maybe I'll find a way to refine the ScreenNavigator api to make these constraints more self-explained